### DISCUSSION TITLE
How to format the SD disk for use in ZuluSCSI?

### DISCUSSION THREAD

**Discussion by mosspa1** [ZuluSCSI/ZuluSCSI-firmware#15] [Q&A] [answered]
Hopefully,  this IS the best place to ask this question.  I've been going down the proverbial "rabbit hole" (sic) for several days now trying to get my two Zulus working. I read through many threads on the Zulu, I've read the GitHub documentation pages as well as the "Zulu Manual", and I've even looked in on a few Amiga clubs. I think I understand how the Zulu devices work, I've had a lot of experience with SCSI since the 90s, and I currently have several SCSI2SD implementations running. My application is in using the Zulus with Kurzweil K2xxx synthesizer/samplers with SCSI interfaces, and I am aware of the apparent compatibility issues with some Kurzweil equipment.  From my perspective, SCSI is SCSI so the host system shouldn't be of much concern since my host Kurzweils (mostly in the 2500 and 26xx series) work fine with any SCSI device I have ever attached to them (and before I believe outright what one person has apparently found specific to only a K2000 and only while running one version of the OS, I'd like to verify it for myself on a wider range of Kurzweil synths).. So, like the SCSI2SD packages, I assume that once everything is properly configured, the Kurzweil OS sees the ZuluSCSI as a SCSI HDD and shouldn't "care" that is is an emulator. In reading from all of the sources I have inferred that everything that is needed to allow the emulator to perform is tied to a filename and, apparently, the contents of the file. In practical application I know what an .img file is and I have played around with them in the past for various reasons. However, I have no idea about how I would go about creating an "empty" one from scratch, and what it would need to include so that it would allow the SD card to function as an emulated 2GB SCSI HDD. Furthermore, I have never heard of or played with an HDA file before, nor do I know how to create one of those, either. To make matters even more different for me, I am, pretty much, tied to Windows systems and I would most like to get access to the card using Windows 7 (that I use for things that might require applications that will no longer run on Windows 10). If it makes any difference, I recently acquired a Mac Mini M1, but I'm guessing that won't do me much more good than any of my Windows machines. Anyway, I know this is probably a very fundamental issue and that I maybe should know more than what I do, but I can't believe that I would be the only person who was flummoxed by the realization that they didn't have the knowledge necessary to enact what appears to be a simple and straightforward set of instructions in order to correctly format the SD card.

Sorry from "just" editing this from a post I apparently made to an unforgiving user group list.  

Thanks in advance,

John

**Comment by PetteriAimonen** [comment, marked as answer]
Hi John,

Here is what I know, hopefully this helps!

- `hda` and `img` are both raw harddrive images, just the name is different. In fact you can name your ZuluSCSI files `HD00.img` if that is easier, the firmware will treat it the same.
- To create empty harddrive image, you need to create a file full of zeros. On Windows there is built-in command you can run in admin command prompt: `fsutil file createnew HD00.hda 2000000000` where the big number is the size of the drive in bytes.
- For convenience I've attached empty 2 GB file that has been zipped.
[2GB_drive.zip](https://github.com/ZuluSCSI/ZuluSCSI-firmware/files/8842193/2GB_drive.zip)
- We haven't yet figured out all the compatibility problems, but we are working on it. Even though SCSI is standardized, there is no bullet-proof way to test that either device fully conforms to the standard. I assume that the developers of hard discs back in the day have also done extensive compatibility testing to fix any problems, and similarly in e.g. SCSI2SD development. ZuluSCSI uses the SCSI2SD command set and benefits a lot of the prior work, but because the hardware interface is different there are some issues that still need figuring out.

**Comment by mosspa1** [comment]
Thank you!!!!  It really is as easy as I thought it might be.  Anyway, if you haven't read my response to the 3.87J incompatibility thread, I just posted a it about a half hour ago.  @chickemeps sent me a 2GB file, similar to the one you sent, and I used it to test the compatibility of a Zulu Mini among all of the Kurzweils in my studio.  Briefly, i didn't have any problems, even with two K2000s I have that are running OS 3.87J.  I don't know how familiar you are with Kurzweils, in general.  But, in the 30 years, or so, I have been using and repairing them, I would never assume a commonality of any problem presented in one machine under one set of circumstances. OS 3.87 was found to be even more robust than I thought it was in my test (for example, it easily resolved the fatal conflict that should have arisen with two different emulators presenting disks at the same SCSI ID address. I accidentally had that happen with the mapping of the Zulu and a resident SCSI2SD in one of the K2000s... the Kurzweil ended up giving precedence to to the Zulu and, afterward, worked flawlessly, albeit with one less drive than it should have had, but both Zulu drive instances still performed admirably, even under those circumstances).

John

    On Monday, June 6, 2022 at 02:07:47 AM EDT, Petteri Aimonen ***@***.***> wrote:  
 
 


Hi John,

Here is what I know, hopefully this helps!
   
   - hda and img are both raw harddrive images, just the name is different. In fact you can name your ZuluSCSI files HD00.img if that is easier, the firmware will treat it the same.
   - To create empty harddrive image, you need to create a file full of zeros. On Windows there is built-in command you can run in admin command prompt: fsutil file createnew HD00.hda 2000000000 where the big number is the size of the drive in bytes.
   - For convenience I've attached empty 2 GB file that has been zipped.   
2GB_drive.zip
   - We haven't yet figured out all the compatibility problems, but we are working on it. Even though SCSI is standardized, there is no bullet-proof way to test that either device fully conforms to the standard. I assume that the developers of hard discs back in the day have also done extensive compatibility testing to fix any problems, and similarly in e.g. SCSI2SD development. ZuluSCSI uses the SCSI2SD command set and benefits a lot of the prior work, but because the hardware interface is different there are some issues that still need figuring out.
